src/controllers/health.check.controller.ts

```diff
@@ -0,0 +1,12 @@
+import { Response, Router, Request } from "express";
+import { handleErrors } from "../middleware/error.handler";
+import { logger } from "../utils/logger";
+
+export const healthCheckController = Router({ mergeParams: true });
+
+async function healthCheck(req: Request, res: Response) {
+    logger.debug(`health check triggered`);
+    await Promise.resolve(res.sendStatus(200));
```

Comment:
I am really unsure about line 9.

Reply:
You could use the await in the calling function to await the result and change this from async or you could just await the res.sendStatus without the Promise other than that looks good.